### DISCUSSION TITLE
The requested resource [/birt-viewer-osgi/frameset] is not available

### DISCUSSION THREAD

**Discussion by shahzad80** [eclipse-birt/birt#1297] [General]
I am trying to set up birt runtime on Rocky Linux 8.8 and Java version is 11.

The welcome page is displaying correctly but when I click on the view example button it displays an error that the frameset is not available.

![http status](https://github.com/eclipse-birt/birt/assets/61746668/517ad3d2-3d2b-4696-bc46-a3c090bf08cb)

can someone guide me on how to remove this error?

I would appreciate a response as soon as possible.

**Comment by speckyspooky** [comment]
It is dependend on your local settings based on your tomcat-server.
The main task would be to check your current server status and which processes are running.

I use not the same system details like you but I think the base checks are the same,
my system/local system details are: BIRT 4.9, Java 17.0.5, Tomcat 9.0.52, Windows 10 (enough RAM)

I will add some screen where your can check the status because it could be that your error is caused based on the wrong URL or that the server process isn't running correctly.

It could be sat there are to many details but I did'n now which status is given on your side.
Before I start, the target is to get the following URL running:
http://localhost:8080/birt/frameset?__report=test.rptdesign&sample=my+parameter

My starting point is that you have installed Tomcat 10 correctly and you copied the birt-folder to Tomcat-folder "webapps".
**I have there the 2 main-folder "birt" and "example":**

![tomcat-00a-folders](https://github.com/eclipse-birt/birt/assets/41593722/0a4b95ef-bf34-46b7-bd5e-586411ee2a45)


After you have started your tomcat-server sucessfully you can check this **at the tomcat-log-folder**.
There you can take a look into the log-files if there an critical error which avoid the process start:

![tomcat-00-log-files](https://github.com/eclipse-birt/birt/assets/41593722/82abad95-9a03-4f5c-92fd-811bd3476bdc)



- ### Check the status of your tomcat:

![tomcat-01-start-screen](https://github.com/eclipse-birt/birt/assets/41593722/0e110b18-7abc-4668-bd17-38627e893363)


- ### Check server status:

![tomcat-02-server-status](https://github.com/eclipse-birt/birt/assets/41593722/15596a2a-a961-4eb2-a3ad-6a37d2755828)


- ### Check the listed applications:

![tomcat-03-applications](https://github.com/eclipse-birt/birt/assets/41593722/f5296b3d-dc9c-4521-a9bb-6ddcb269286d)


- ### Check the "example" link of birt:

![tomcat-04-viewer-example](https://github.com/eclipse-birt/birt/assets/41593722/a5ed7098-4748-4b46-b296-ed8fa549de89)


- ### Check the usage based on your "birt"-folder - my example it is called "birt":

![tomcat-05-birt-viewer-own](https://github.com/eclipse-birt/birt/assets/41593722/0226663e-cb76-4f2b-aa9d-2bb5e7392ffa)


The question is to which point you can go ahead till you see the first problem.

**Reply by shahzad80** [reply]
Thank you for your prompt and detailed reply.

this is right that your system spec and OS details are changed from mine system.
I am using RockLinux 8.7 BIRT 4.9, Java 11.0.19, Tomcat 10.0.21

This is my system details
![rocky+tomcat](https://github.com/eclipse-birt/birt/assets/61746668/289e03fe-9147-4a7b-93e8-7596f31dc479)

I have just copied the runtime BIRT folder name birt-viewer-osgi into this path /opt/tomcat/webapps
under this folder, there was a directory name WebViewerExample and I rename it with the name birt-viewer then I tried to access the index.jsp which was present in the birt-viewer and then set the path of this file in tomcat. service file.

![systemfile](https://github.com/eclipse-birt/birt/assets/61746668/70b5fb16-4131-473e-a78d-d7ead4e5f990)

when I typed this URL into the browser it worked fine.
like this http://172.30.100.233:8080/birt-viewer-osgi/birt-viewer/ 
![birt-viewer](https://github.com/eclipse-birt/birt/assets/61746668/2a760880-adf0-411b-9e24-8d3a9aeed863)

as I clicked on the view example button it displayed an error in the frameset.
![birt-error](https://github.com/eclipse-birt/birt/assets/61746668/8c0a39b4-2a98-4cfc-98f2-1ae5b9c9201f)

I can access the Tomcat server page 
![tomcat](https://github.com/eclipse-birt/birt/assets/61746668/75d8daef-184e-415c-9461-763701095bfe)

as per my understanding, this URL http://172.30.100.233:8080/birt-viewer-osgi/birt-viewer/ opens this index.jsp file that I highlighted in the image and you can see that there are two files present for testing and launching the first title page of the report called test.rptdesign
![index](https://github.com/eclipse-birt/birt/assets/61746668/51d884e7-eeb8-4197-9311-5e8ce73d1b00)

Thanks again and really appreciate your other quick and detailed response.

**Reply by shahzad80** [reply]
I am attaching a screenshot of the Tomcat server log so you can find any clue that where am I making a mistake 

![logs](https://github.com/eclipse-birt/birt/assets/61746668/7cc0f644-114b-47ae-ab63-44950329466c)

Thanks.

**Comment by speckyspooky** [comment]
I can see 2 404 errors in your screen and the last of the error ist the "frameset?___..."
![Unbenannt](https://github.com/eclipse-birt/birt/assets/41593722/7435cf22-c3aa-4925-8d38-bdcef46f7237)

The good think is that the BIRT-eclipse starting page is displaid so we know it is not complettely wrong and crashed.
The 404-error here means that the server could'n find your target.

You have wiritten that your URL e.g. "http://172.30.100.233:8080/birt-viewer-osgi/birt-viewer/" was successful,
what appears if you need this URL and create the URL like:
"http://localhost:8080/**birt-viewer-osgi/birt-viewer**/frameset?__report=test.rptdesign&sample=my+parameter"
(based on your example).

Perhaps you will get a result.

**Reply by shahzad80** [reply]
Thank you again for getting back to me so quickly.

I tried to create and access a link that you suggested in your reply but it is still throwing the same error as it was displaying before like a Birt frameset error.

![link](https://github.com/eclipse-birt/birt/assets/61746668/3c2f665a-f6e4-49a4-8569-46400cb4b2f3)

![localhost](https://github.com/eclipse-birt/birt/assets/61746668/eb70d8d9-e1e9-4595-92a6-a3651355c50f)

I checked and viewed the log and it showed a 404 error which you explained in your message. I am attaching complete logs in the message.

![birt-logs](https://github.com/eclipse-birt/birt/assets/61746668/d3affdd8-2457-492f-b80c-3309abc38472)
![birt-logs-1](https://github.com/eclipse-birt/birt/assets/61746668/3c643fd7-3156-4ce0-857a-74ceec7d7fd1)

**Reply by shahzad80** [reply]
I checked all folder permission and its main directory permission maybe there is an issue related to the permission of all directory contents belonging to user Tomcat.

everything is ok regarding the permission.

I deployed a sample war file into the Tomcat server and its contents are displaying well.

![sample](https://github.com/eclipse-birt/birt/assets/61746668/a178ff2f-f573-4ec4-bb3d-d73b4fca5489)

but when I click on the serverlet link it displays an error.

![sample-error](https://github.com/eclipse-birt/birt/assets/61746668/a4b1e156-0600-42ca-ae99-018693d216f9)

**Reply by speckyspooky** [reply]
The difference on your last screen is that the URL isn't the same.
Therefore you have this effect "hello.jsp" is found but not the target source "hello".
By default he will try to fetch a file with the extention "hello.html".

The HTTP-404 error means realy that the server is running but not able to finde the source-files.

I have currently only one idea, could you please make a screen of your folder structure based with "ls" of the webapps-folder.
By default the URL-part "<loacalhost:8080>" is the reference to tomcat-webapps-folder.

The path behind is your local folder structure and local folder names - example:
![tomcat-path](https://github.com/eclipse-birt/birt/assets/41593722/03fcd6c2-e819-41b3-b887-a1e597a0cf83)


So I think your are on the right way and the default configuration is correct.
The correct path is the way. As a test you can also create a very very small HTML-file on your own which you can place on your "example" folder and also "birt-folder" to fetch it later with the URL call.

The important thing is that the path like a folder path is correct and afterwards the source-file (HTML-file) or later the "frameset" (this not a page) cann be called.

**Reply by shahzad80** [reply]
Thank you for your reply.

The URL is not the same because you mentioned in your message that creates a URL link like localhost:8080\birt-viewer and before I was using a URL with IP like http://IP-Address:8080/birt-viewer.

but I didn't notice any differences in the results by accessing both methods of mentioning IP or localhost to access the birt_viewer frameset
report.

they were displaying the same error after clicking on the link birt-viewer example link.

this is the local structure of Tomcat webapps folders.

![image](https://github.com/eclipse-birt/birt/assets/61746668/956fa7c4-8f88-4f09-a7a2-c589d78c25a3)

when I do cd into the example folder there is an index.html file present which you have highlighted in your image and you have executed this index.html file through localhost:8080/examples/index.html

this is my local example folder

![local](https://github.com/eclipse-birt/birt/assets/61746668/9d4a54f7-6aa0-4e32-8226-3ac611060293)

I tried to get the index.html file from /opt/tomcat/webapps/examples folder via URL link from 

**http://172.30.100.233:8080/examples/index.html**
and
**elinks http://localhost:8080/examples/index.html**

they all are working fine.

![image](https://github.com/eclipse-birt/birt/assets/61746668/83953afc-af26-4543-9b34-4c0ce5ea111d)

![image](https://github.com/eclipse-birt/birt/assets/61746668/85e01ef0-dd18-41c1-a303-dde670901cd8)

![image](https://github.com/eclipse-birt/birt/assets/61746668/eab0bc8f-e45e-49ae-87c0-35eef64ec7ac)

![image](https://github.com/eclipse-birt/birt/assets/61746668/0a74ea46-d6a7-4b62-8dfa-f12486b46323)

![image](https://github.com/eclipse-birt/birt/assets/61746668/0bf35e2e-71ba-4127-9602-5d6d36d110b7)

furthermore, to your understanding, I am doing cd into this parent folder and there is another folder name birt-viewer under this folder there is an index.jsp file if I am not mistaken so when I write URl on the browser like 

http://172.30.100.233:8080/birt-viewer-osgi/birt-viewer/

it calls that index.jsp file which resides in this path /opt/tomcat/webapps/birt-viewer-osgi/birt-viewer

![image](https://github.com/eclipse-birt/birt/assets/61746668/711e1731-fe9d-4f4d-8ccd-33977960b7a0)

The welcome page of birt reporting displays fine.

after clicking on the view example button server tries to fetch a test.rptdesign file which is shown in the URL that gets server parameter to access the default title page of the birt viewer I would like to bring notice to you that test.rptdesign file is present in this folder so I didn't come to know why server is not able to find this test file.

![image](https://github.com/eclipse-birt/birt/assets/61746668/41156295-5d4b-4cfc-bf1d-cba2fdf5ccf7)

![image](https://github.com/eclipse-birt/birt/assets/61746668/b5f463eb-cc4a-421c-8ed5-791bbaa62727)

**Reply by shahzad80** [reply]
one thing more in Linux system is we need to define BIRT-Home path in the tomcat service file and I set the path of birt-viewer folder in it.

could you please check out parameters that I declare in this service file?

![image](https://github.com/eclipse-birt/birt/assets/61746668/fc70a221-e2a7-4960-aa65-e9b4f78cd778)

I think in Windows operating system we don't need to set BIRT-HOME variable in the system.

**Reply by shahzad80** [reply]
really appreciate your prompt and quick reply again.

Thank you again for your great support.

**Reply by speckyspooky** [reply]
The BIRT_HOME-variable isn't set on Windows.

But I figured out a difference this could be given through the Windows-handling but perhaps you can test it.
I downloaded the OSGI-version of BIRT and installed it like your path ".../webapps/birt-viewer-osgi/birt-viewer" but this configuration doesn't work on my side.
So moved the content of "birt-viewer" directly under the osgi-folder then my path was ".../webapps/birt-viewer-osgi".
I removed the old folder and after a tomcat restart, I was able to call the osgi-version:

![tomcat-birt-viewer-osgi](https://github.com/eclipse-birt/birt/assets/41593722/b7e90df2-20ec-495c-b882-65b4a583f607)

URL: http://localhost:8080/birt-viewer-osgi/frameset?__report=test.rptdesign&sample=my+parameter

**Reply by shahzad80** [reply]
Thank you for your reply and great support.

I have tried the same way as you suggested I rename the birt-viewer to birt-viewer-bck and copy all its contents into 
/opt/tomcat/webapps/birt-viewer-osgi directly

![image](https://github.com/eclipse-birt/birt/assets/61746668/c152fd04-bf92-47bc-b9df-ad6d03dc9aaf)

but it stopped working and even it is not able to get birt home page.

I tried it both ways like 

http://172.30.100.233:8080/birt-viewer-osgi
and
http://172.30.100.233:8080/webapps/birt-viewer-osgi

it's returning 404 error

![image](https://github.com/eclipse-birt/birt/assets/61746668/d58421e2-c8c2-4b7e-a81b-8baac7c6ca5d)
![image](https://github.com/eclipse-birt/birt/assets/61746668/4570d700-212d-4046-8727-38b0be8a462b)

apart from this I also can not be able to call other files that exist in this folder like notice.html, about.html, epl-2.0.html

![image](https://github.com/eclipse-birt/birt/assets/61746668/fac45705-5811-4f14-905d-8d6989f798c9)
![image](https://github.com/eclipse-birt/birt/assets/61746668/c2929935-a4eb-4d4d-a02f-c431eb4426c3)

**Reply by shahzad80** [reply]
is it necessary to keep a folder under the name of birt_viewer then it would work as I copied all contents of birt-viewer folder into birt-viewer-osgi folder?

it stopped working and another strange thing is that birt.war file is located in birt-viewer-osgi folder which is responsible for running all files but it's not working.

am I making any other mistake?

Thanks again for your patience and great support in bearing all my queries.

**Reply by shahzad80** [reply]
I removed all these files from birt-viewer-osgi folder. so by copying all of the contents of birt-viewer folder into parent directory(birt-viewer-osgi) is not workable in the Linux system I think.

![image](https://github.com/eclipse-birt/birt/assets/61746668/f751972c-e310-43fa-8379-5989f9e2309f)

it starts working again and the home page is displaying on the browser.

![image](https://github.com/eclipse-birt/birt/assets/61746668/1da9c639-5821-4e26-a0e9-5ecf51cdf931)

**Reply by shahzad80** [reply]
that was an actual link below mentioned

https://stackoverflow.com/questions/42665624/issues-in-birt-deployment-in-tomcat

https://bugs.eclipse.org/bugs/show_bug.cgi?id=497356

I followed the steps mentioned in this link but had no luck.

I deleted files that are named under ECLIPSE_.RSA
ECLIPSE_.SF.

still having the same screen frameset is not available.

one thing I noticed when I type this URL http://172.30.100.233:8080/birt-viewer-osgi/birt-viewer-bck/ then it shows birt Home Page but when I click on the view example link it excludes the folder birt-viewer-bck and create a link like

http://172.30.100.233:8080/birt-viewer-osgi/frameset?__report=test.rptdesign&sample=my+parameter

what I want to say is that it tries to fetch test.rptdesign into birt-viewer-osgi folder however this file present on this path 
/opt/tomcat/webapps/birt-viewer-osgi/birt-viewer-bck although birt has to lookup this file in birt-viewer-bck folder.

i didn't understand that thing too.

**Reply by speckyspooky** [reply]
It isn't necessary to have the folder name birt_viewer.
Have you changed the BIRT_HOME on your tests?

The folder "webapps" is always the root and not content of the target URL.

**Reply by shahzad80** [reply]
yes I changed it before copying all files into /opt/tomcat/webapps/birt-viewer-osgi

**Reply by shahzad80** [reply]
issue is that when i set a path directly of Environment=BIRT_HOME=/opt/tomcat/webapps/birt-viewer-osgi.
it does not work even I can not be able to browse birt homepage

**Reply by shahzad80** [reply]
does it require changing the port of Tomcat. it is listening on port no 8080 and I think birt runtime does not have concerns with that

![image](https://github.com/eclipse-birt/birt/assets/61746668/1d6ba4a0-4aed-4774-9bdf-da86833a7b13)

**Reply by shahzad80** [reply]
![image](https://github.com/eclipse-birt/birt/assets/61746668/c3c8094a-496e-45d8-a62b-6b9ce43dfb74)

I have declared BIRT-HOME in the system file but no contents were accessible through this path

**Reply by speckyspooky** [reply]
It is a little bit hard to assist you with further ideas. The only thing is that it isn't required for BIRT to use an oher prot-address because the web application run in the context of your tomcat so all is fine of the port.

Your note with the birt.war is a little bit strange because if you copy the Example and rename the folder you shouldn't have the birt.war into this folder - remove it from there.

You can giv Tomcat a try to deploy the application by him self.
For that copy only the BIRT.war file to the "webapp"-folder and rename it to e.g. "BIRT-next-try.war".
If you restart the tomcat then the folder "BIRT-next-try" will be created through Tomcat and you can see the new version in your Tomcat-manager.

I have also read your reference to the signature issue. But this shoudl be solved with the last comment of end of 2017.

**Reply by shahzad80** [reply]
Thank you for your reply.

I gave it a try to copy Birt.war into Birt-viewer folder and also kept a copy Birt,war in web apps folder so might that work.

but no luck and I just deleted that file again then restart Tomcat server but problem remained same.

is it really complicated to configure Birt on a Linux machine or do I have some mystery regarding this issue : )

I installed and configured same things in VM and it has Redhat 8 but it also giving me same issue related to birt frameset.

**Reply by shahzad80** [reply]
Hello,

How are you doing?

got any luck regarding this issue Sir?

Thanks.

**Reply by shahzad80** [reply]
![image](https://github.com/eclipse-birt/birt/assets/61746668/fe177341-638c-4a5c-ac28-d1af8a72425d)

This is my server status and if you see on this image there is birt-osgi appearing on the server but again when I try to access from there it comes up with the same issue.

![image](https://github.com/eclipse-birt/birt/assets/61746668/f168e261-6b40-4714-b6d4-db1036da0e49)

**Reply by speckyspooky** [reply]
Currently I haven't any new ideas or information. On  my side I haven't Linux-server available thefore I cannot directly compare with your setup.

Have you tried your retest with the latest version BIRT-osgi 4.13 & Java 11?
Because in one of your screens is BIRT 4.9.0 listed. Therefore it would make a try to use the released version BIRT 4.13.

Could you see some other issue in the standard error-log file of your wildfly or on your catalina-log-file?
Perhaps some additional hints on it...

**Reply by shahzad80** [reply]
The same behavior happened after copying Birt 4.13 with Java 11 and 17 versions I tried this latest version first on the VM then I later moved to configure Birt 4.9 on a physical machine.

this is a log entry of Catalina.out file. but in this file, there is no specific information to get a clear picture to resolve this issue.

![image](https://github.com/eclipse-birt/birt/assets/61746668/09849d43-395a-4581-8e46-2d570472f267)

![image](https://github.com/eclipse-birt/birt/assets/61746668/e8416d9d-5a8b-43b4-9364-797943341ed6)

![image](https://github.com/eclipse-birt/birt/assets/61746668/a51b1b6d-4e61-4ba4-a153-1f1e2907dac3)

**Reply by shahzad80** [reply]
can you do remote access to my machine if you have spare time to check the settings on the VM?

you can join online session via any desk if you are comfortable with that have no other plans tomorrow or on Sunday.

Thanks.

**Comment by shahzad80** [comment]
today I downloaded old version of BIRT birt-runtime-4_2_2 but encountered the same error that was coming from the latest versions.

permissions are aligned correctly on runtime folder but really don't know exactly at what point or where am i making a mistake.

![image](https://github.com/eclipse-birt/birt/assets/61746668/6715fc2f-5b97-4c6a-baae-c7927dc923a2)

**Comment by claesrosell** [comment]
Hi.

I would just like to say that the key difference between Tomcat 10 and Tomcat 9 is that Tomcat 10 is only compatible with the new "jakarta" namespace and Tomcat 9 is compatible with the old "javax" namespace.

BIRT still relays on the old "javax" namespace, hence it is incompatible with Tomcat 10. You should use Tomcat 9 instead, hopefully, that will solve your problem.

**Reply by shahzad80** [reply]
Thank you for your reply.

I will try your suggestion and let you know whether it works or not.

Thanks.

**Reply by shahzad80** [reply]
![image](https://github.com/eclipse-birt/birt/assets/61746668/a5ae77b4-44e7-458a-ae1c-fca20c5d0bf0)

![image](https://github.com/eclipse-birt/birt/assets/61746668/191915c5-a3a5-46a3-af07-2c6e77a1b5ea)

when I clicked on the Birt link it showed me Birt homepage properly

![image](https://github.com/eclipse-birt/birt/assets/61746668/5e95389a-3322-4ed6-a06f-45323acd68d4)

after that, I clicked on the view example link status of error changed but error code was the same like 404 but message in staus displayed changed like:

The origin server did not find a current representation for the target resource or is not willing to disclose that one exists.

![image](https://github.com/eclipse-birt/birt/assets/61746668/15a902e1-80ad-43e3-bae4-0b7d6f3805d8)

these are the contents of my web apps folder

![image](https://github.com/eclipse-birt/birt/assets/61746668/cf48168f-7fce-4bcd-9f0c-96a01d251aa8)

**Reply by shahzad80** [reply]
service file parameters in tomcat service file in the linux

![image](https://github.com/eclipse-birt/birt/assets/61746668/fbff983c-13dc-4a85-960f-d53b45571c39)

**Reply by speckyspooky** [reply]
Please try it again based on our reserach before and enter the URL directly
with your current setup of Tomcat 9, BIRT 4.9, Java 11:

http://<your IP-address & port>/birt/frameset?__report=test.rptdesign&sample=my+parameter
e.g.:
http://localhost:8080/birt/frameset?__report=test.rptdesign&sample=my+parameter

**Reply by shahzad80** [reply]
Thank you for your reply.

I really appreciate your research and the great support that you constantly give me.

at first time when I click on view example link it gave me this message and page view and showed error 500.

I tried to through this link on my local IP 192.168.1.10

****http://192.168.1.10:8080/birt/frameset?__report=test.rptdesign&sample=my+parameter****

![image](https://github.com/eclipse-birt/birt/assets/61746668/2711f4f5-136d-4bf1-ac24-685a310e4c0f)

after that I tried your mentioned like on my vm

**http://localhost:8080/birt/frameset?__report=test.rptdesign&sample=my+parameter**

![image](https://github.com/eclipse-birt/birt/assets/61746668/f8ca331f-86c3-4525-adce-99ea1198a635)

**Reply by shahzad80** [reply]
if you look at on server state image which I shared with you earlier of Tomcat 9 server. The state of birt and view-example is true it shows that they are configured correctly but server is not able to fetch the desired link or maybe it is some Java misconfiguration causing that issue.

I shared an image file that tells some Java files are not compatible with that link which I understand from this message.

**Reply by claesrosell** [reply]
The error message suggests that you have encountered bug #941, it was discussed and a workaround presented in #940

Birt 4.9 had a lot of issues on release. Newer versions is better.

**Reply by shahzad80** [reply]
Thank you again for your reply.

should I try Birt latest version 4.13 on Tomcat 9?

I tried on Tomcat 10 but it was not worked out.

**Reply by claesrosell** [reply]
Yes. Go back to  4.13 but use Tomcat 9 this time.
I have not used it myself but it should be the best release of BIRT.

**Reply by shahzad80** [reply]
Go to : "apache-tomcat-9.0.29/webapps/birt-viewer/WEB-INF/lib"
Unzip the "javax.xml.rpc-api-1.1.2.jar" file that are inside the javax.xml.rpc_1.1.2.v202203170910.jar
Remove the javax.xml.rpc_1.1.2.v202203170910.jar
Restart Tomcat

in my case above mentioned file that you pointed out is not present in my folder

**/opt/tomcat/webapps/birt/WEB-INF/lib**
I searched into this folder even I searched entire Tomcat folder but there is no file under this name **javax.xml.rpc_1.1.2.v202203170910.jar**

![image](https://github.com/eclipse-birt/birt/assets/61746668/59b6b64f-26dd-4de4-bed9-f14fa49cf8fe)

![image](https://github.com/eclipse-birt/birt/assets/61746668/2042aec1-7880-4e15-853b-84d83ad911f8)

**Reply by shahzad80** [reply]
ok I am working on it and share the results with you in a short while

**Reply by shahzad80** [reply]
![image](https://github.com/eclipse-birt/birt/assets/61746668/aa82f498-79ca-4ff5-a528-d745dcc98c02)

it worked without any issues.

Thank you for your great support both of you **Thomas Gutmann** and **Claes Rosell**

you guys are great and without your continued support, I was not able to do this task.

once again Thanks a lot.

can I connect you both of you people on LinkedIn or any other platform?
